src/Epsagon.Dotnet.Lambda/EpsagonGeneralHandler.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Epsagon.Dotnet.Core;
+using Epsagon.Dotnet.Tracing.Legacy;
+using Epsagon.Dotnet.Tracing.OpenTracingJaeger;
+using OpenTracing;
+using OpenTracing.Util;
+
+namespace Epsagon.Dotnet.Lambda
+{
+    public class EpsagonGeneralHandler
+    {
+        public static void Handle(Action clientFn, [CallerMemberName] string methodName = "")
+        {
+            if (Utils.CurrentConfig == null || Utils.CurrentConfig.IsEpsagonDisabled)
+            {
+                clientFn();
+                return;
+            }
+
+            try
+            {
+                using (var scope = CreateRunner(methodName))
+                {
+                    ExecuteClientCode(clientFn, scope);
+                }
+            }
+            finally { CreateTraceAndSend(); }
+        }
+
+        public static T Handle<T>(Func<T> clientFn, [CallerMemberName] string methodName = "")
+        {
+            if (Utils.CurrentConfig == null || Utils.CurrentConfig.IsEpsagonDisabled) return clientFn();
+
+            T result;
+            try
+            {
+                using (var scope = CreateRunner(methodName))
+                {
+                    result = ExecuteClientCode(clientFn, scope);
+                    scope.Span.SetDataIfNeeded("aws.lambda.return_value", result);
+                }
+            }
+            finally { CreateTraceAndSend(); }
+            return result;
+        }
+
+        public static async Task<T> Handle<T>(Func<Task<T>> clientFn, [CallerMemberName] string methodName = "")
+        {
+            if (Utils.CurrentConfig == null || Utils.CurrentConfig.IsEpsagonDisabled) return await clientFn();
+            T result;
+            try
+            {
+                using (var scope = CreateRunner(methodName))
+                {
+                    result = await ExecuteClientCode(clientFn, scope);
+                    scope.Span.SetDataIfNeeded("aws.lambda.return_value", result);
+                }
+            }
+            finally { CreateTraceAndSend(); }
+            return result;
+        }
+
+        private static void CreateTraceAndSend()
+        {
+            var trace = EpsagonConverter.CreateTrace(JaegerTracer.GetSpans());
+            EpsagonTrace.SendTrace(trace);
+            JaegerTracer.Clear();
+        }
```

Comment:
why? let jaeger tracer deal with the sending, you just need to create spans (no need for the finally in the above functions either)